File: e2e/cypress-javascript/interactions/steps/LanguageSwitcherStep.js
const { step } = require('@/decorators/step-decorator');
const { LanguageSwitcherDropdown } = require('@/interactions/pages/LanguageSwitcherDropdown');

class LanguageSwitcherStep {
  constructor() {
    this.dropdown = new LanguageSwitcherDropdown();
    this.selectLanguage = step("Select language '{language}'", this._selectLanguage.bind(this));
  }

  _selectLanguage(language) {
    this.dropdown.dropdown.click();
    this.dropdown.items.contains(language).click();
    this.dropdown.dropdownValue.should('have.text', language);
  }
}

module.exports = { LanguageSwitcherStep };

File: e2e/capybara-ruby/interactions/steps/language_switcher_step.rb
# frozen_string_literal: true

require_relative '../../decorators/step_decorator'
require_relative '../../support/capybara_helpers'
require_relative '../pages/language_switcher_dropdown'

class LanguageSwitcherStep
  extend StepDecorator::Methods
  include CapybaraHelpers

  def initialize
    @dropdown = LanguageSwitcherDropdown.new
  end

  step "Select language '{language}'"
  def select_language(language)
    @dropdown.dropdown.click
    @dropdown.items.find { |item| item.text == language }.click
    expect(@dropdown.dropdown_value).to have_text(language)
  end
end
